### PR TITLE
Feature/extend ci ubuntu 16.04

### PR DESCRIPTION
**Pull request by jafet-chaves**
This extends the CI to run on Ubuntu 16.04 container

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -11,8 +11,13 @@ on:
 
 jobs:
   build_autotools:
-    runs-on: ubuntu-18.04
-    container: ridgerun/r2inference:v0.1.5
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+       os: [ubuntu-18.04]
```
Comment (rrcarlosrodriguez) [inline .github/workflows/main.yml:17]: Shouldn't we add ubuntu-16.04 too?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -59,13 +64,18 @@ jobs:
...
         backend='tensorflow' input-layer='input' output-layer='InceptionV1/Logits/Predictions/Reshape_1' \
         labels=$ROOT/imagenet_labels.txt crop=false overlay=true filter=-1 ! \
         fakesink silent=false sync=false async=false -v
   build_meson:
-    runs-on: ubuntu-18.04
-    container: ridgerun/r2inference:v0.1.5
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+       os: [ubuntu-18.04]
```
Comment (rrcarlosrodriguez) [inline .github/workflows/main.yml:75]: Same here

Comment (jafet-chaves) [reply .github/workflows/main.yml:75]: Check reply on the same subject here: https://github.com/RidgeRun/r2inference/pull/83#discussion_r430042565

File: .github/workflows/main.yml (hunk 1 above)
Comment (michaelgruner) [reply .github/workflows/main.yml:17]: I agree, remove this from the matrix.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -11,8 +11,13 @@ on:
...
+       container: ["ridgerun/r2inference:v0.1.5", "ridgerun/r2inference-ubuntu-16.04:v0.3.0"]
+    container:
+      image: ${{ matrix.container }}
     env:
       PREFIX: /usr/lib/x86_64-linux-gnu/
       CFLAGS: -Werror
```
Comment (michaelgruner) [inline .github/workflows/main.yml:23]: CXXFLAGS

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -59,13 +64,18 @@ jobs:
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+       os: [ubuntu-18.04]
+       container: ["ridgerun/r2inference:v0.1.5", "ridgerun/r2inference-ubuntu-16.04:v0.3.0"]
+    container:
+      image: ${{ matrix.container }}
     env:
       PREFIX: /usr
       CFLAGS: -Werror
```
Comment (michaelgruner) [inline .github/workflows/main.yml:81]: CXXFLAGS

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -11,8 +11,13 @@ on:
 
 jobs:
   build_autotools:
-    runs-on: ubuntu-18.04
-    container: ridgerun/r2inference:v0.1.5
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+       os: [ubuntu-18.04]
+       container: ["ridgerun/r2inference:v0.1.5", "ridgerun/r2inference-ubuntu-16.04:v0.3.0"]
```
Comment (michaelgruner) [inline .github/workflows/main.yml:18]: Please make a tag for the `ridgerun/r2inference:v0.1.5` image to have a consistent name.

**Pull request merged**